climakitae/util/utils.py

```diff
@@ -520,3 +520,40 @@ def summary_table(data):
         df = df.sort_values(by=["year"])
 
     return df
+
+
+def convert_to_local_time(data, selections, lat, lon) -> xr.Dataset:
+    """
+    Converts the inputted data to the local time of the selection.
+    """
+    # 1. Find the other data
+    start, end = selections.time_slice
+    tz_selections = copy.copy(selections)
+    tz_selections.time_slice = (
+        end + 1,
+        end + 1,
```

Comment:
Test the case where you are extracting data at the end of the time slice (2100).

Reply:
2015 is end of historical

Reply:
> Temporal range of each dataset:
> historical_climate_range_wrf = (1980, 2015)
> historical_climate_range_loca = (1950, 2015)
> historical_climate_range_wrf_and_loca = (1981, 2015)
> historical_reconstruction_range = (1950, 2022)
> ssp_range = (2015, 2100)

Reply:
At the end of the time slice, don't bother getting the additional data. Just adjust the timezones.

Reply:
Test with daily/monthly data too, instead of just hourly.

Reply:
Try to test slicing on non-year slice. Instead of (2051, 2051), do a timestamp.

Reply:
> Try to test slicing on non-year slice. Instead of (2051, 2051), do a timestamp.

I would deprioritize this as the last item on the list of getting the timezone conversion functionality working, and would say it's not necessary at the present time. None of our notebooks at present have a finer non-year slice for data pulling.